trajopt_optimizers/trajopt_sqp/src/trust_region_sqp_solver.cpp

```diff
@@ -88,6 +93,14 @@ void TrustRegionSQPSolver::solve(const QPProblem::Ptr& qp_problem)
     // Convexification loop
     for (int convex_iteration = 0; convex_iteration < 100; convex_iteration++)
     {
+      double elapsed_time = std::chrono::duration<double, std::milli>(Clock::now() - start_time).count() / 1000.0;
+      if (elapsed_time > params.max_time)
+      {
+        CONSOLE_BRIDGE_logInform("Elapsed time %f has exceeded max time %f", elapsed_time, params.max_time);
+        status_ = SQPStatus::TIME_LIMIT;
+        break;
+      }
+
       if (stepSQPSolver())
         break;
 
```

Comment:
Maybe also move the iteration limit check (line 107) to before stepSQPSolver(), otherwise in a hypothetical case where the very last solve step succeeds, but the iteration limit is reached at the same time, the whole solve process fails.